### ISSUE TITLE
UserspaceEmulator+libcompress: SoftMMU::write32: No region for @ 0x0ffff77c

### ISSUE THREAD

**Issue by bcoles** [SerenityOS/serenity#5778]
This was working a couple of days ago. Presumably one of the recent libcompress commits broke it.

Reproduce:
* `ue Cube` ; or
* `ue Fire`

Output:
```
UserspaceEmulator(29): interpreter: /usr/lib/Loader.so
SoftMMU::write32: No region for @ 0x0ffff7fc
=={29}==    0xb2592fc2  [libcompress.so]: Compress::CanonicalCode::from_bytes(AK::Span<unsigned char const>) +0x22 (Deflate.cpp:68)
=={29}==    0xb259525e  [libcompress.so]: Compress::DeflateDecompressor::decode_codes(Compress::CanonicalCode&, AK::Optional<Compress::CanonicalCode>&) +0x7be (Deflate.cpp:393)
=={29}==    0xb2596f21  [libcompress.so]: Compress::DeflateDecompressor::read(AK::Span<unsigned char>) [clone .part.0] +0xc61 (Deflate.cpp:250)
=={29}==    0xb2597d08  [libcompress.so]: Compress::DeflateDecompressor::decompress_all(AK::Span<unsigned char const>) +0x248 (Span.h:89)
=={29}==    0x8dbaae0d  [libgfx.so]: Gfx::decode_png_bitmap(Gfx::PNGLoadingContext&) +0x9d (Optional.h:127)
=={29}==    0x8dbac3b9  [libgfx.so]: Gfx::load_png_impl(unsigned char const*, unsigned long) +0xc9 (PNGLoader.cpp:817)
=={29}==    0x8dbac766  [libgfx.so]: Gfx::load_png(AK::StringView const&) +0xd6 (Atomic.h:266)
=={29}==    0x8db657df  [libgfx.so]: Gfx::Bitmap::load_from_file(AK::StringView const&, int) +0x102f (Bitmap.cpp:175)
=={29}==    0x1a6d9d72  [libgui.so]: GUI::Icon::default_icon(AK::StringView const&) +0xb2 (Atomic.h:266)
=={29}==    0x86636099  [/bin/Fire]: main +0x2b9 (Atomic.h:266)
=={29}==    0x86636265  [/bin/Fire]: _start +0x45 (crt0_shared.cpp:60)
=={29}==    0x08028aa2
=={29}==    0x0800155b
UserspaceEmulator(29): USERSPACE(29) ASSERTION FAILED: false
../Userland/DevTools/UserspaceEmulator/SoftMMU.cpp:227
ASSERTION FAILED: false

```

Log:

<details>
<p>

```
UserspaceEmulator(29): interpreter: /usr/lib/Loader.so
UserspaceEmulator(29): USERSPACE(29) ASSERTION FAILED: false
../Userland/DevTools/UserspaceEmulator/SoftMMU.cpp:227
[#0 (UE) Fire(29:29)]: 0x00000000  (?)

[#0 (UE) Fire(29:29)]: 0xc0367c46  Kernel::Process::crash(int, unsigned int, bool) +0x3e2
[#0 (UE) Fire(29:29)]: 0xc038c3db  Kernel::Process::sys$abort() +0x3d
[#0 (UE) Fire(29:29)]: 0xc0387ab7  syscall_handler +0x18a4
[#0 (UE) Fire(29:29)]: 0xc03861db  syscall_asm_entry +0x31
[#0 (UE) Fire(29:29)]: Process regions:
[#0 (UE) Fire(29:29)]: BEGIN       END         SIZE        ACCESS  NAME
[#0 (UE) Fire(29:29)]: 01a01000 -- 01b00fff 00100000 RW  T  Stack (Main thread)
[#0 (UE) Fire(29:29)]: 01b02000 -- 01b02fff 00001000 R XS C Signal trampoline
[#0 (UE) Fire(29:29)]: 01b04000 -- 01b04fff 00001000 RW S   
[#0 (UE) Fire(29:29)]: 01b06000 -- 01b06fff 00001000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01b0a000 -- 01b0bfff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 01b0d000 -- 01b0dfff 00001000 RW     mmap
[#0 (UE) Fire(29:29)]: 01b10000 -- 01b1ffff 00010000        malloc: ChunkedBlock(256)
[#0 (UE) Fire(29:29)]: 01b21000 -- 01b24fff 00004000 R  S   /res/fonts/CsillaBold10.font
[#0 (UE) Fire(29:29)]: 01b26000 -- 01b26fff 00001000 R  S   /res/icons/16x16/app-fire.png
[#0 (UE) Fire(29:29)]: 01b28000 -- 01b28fff 00001000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01b2b000 -- 01b2bfff 00001000 RW     
[#0 (UE) Fire(29:29)]: 01b2d000 -- 01b2dfff 00001000 R      mmap
[#0 (UE) Fire(29:29)]: 01b30000 -- 01b3ffff 00010000 RW     malloc: ChunkedBlock(64)
[#0 (UE) Fire(29:29)]: 01b41000 -- 01b44fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01b4b000 -- 01b4cfff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01b50000 -- 01b5ffff 00010000 RW     malloc: ChunkedBlock(8)
[#0 (UE) Fire(29:29)]: 01b61000 -- 01b65fff 00005000 R  S   /res/fonts/CsillaBold12.font
[#0 (UE) Fire(29:29)]: 01b6b000 -- 01b6bfff 00001000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01b6d000 -- 01b6dfff 00001000 R  S   libsystem.so: .text
[#0 (UE) Fire(29:29)]: 01b70000 -- 01b7ffff 00010000 RW     malloc: ChunkedBlock(128)
[#0 (UE) Fire(29:29)]: 01b81000 -- 01b85fff 00005000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01b89000 -- 01b89fff 00001000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01b8b000 -- 01b8bfff 00001000 RW     mmap
[#0 (UE) Fire(29:29)]: 01b8d000 -- 01b8dfff 00001000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01b90000 -- 01b9ffff 00010000 RW     malloc: ChunkedBlock(32)
[#0 (UE) Fire(29:29)]: 01ba1000 -- 01ba4fff 00004000 R  S   /res/fonts/CsillaRegular10.font
[#0 (UE) Fire(29:29)]: 01ba6000 -- 01ba9fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01bad000 -- 01badfff 00001000 RW     mmap
[#0 (UE) Fire(29:29)]: 01bb0000 -- 01bbffff 00010000 RW     malloc: ChunkedBlock(64)
[#0 (UE) Fire(29:29)]: 01bc1000 -- 01bc5fff 00005000 R  S   /res/fonts/CsillaRegular12.font
[#0 (UE) Fire(29:29)]: 01bc7000 -- 01bcafff 00004000 R  S   /res/fonts/KaticaBold10.font
[#0 (UE) Fire(29:29)]: 01bcd000 -- 01bcdfff 00001000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01bd0000 -- 01bdffff 00010000 RW     malloc: ChunkedBlock(8184)
[#0 (UE) Fire(29:29)]: 01be1000 -- 01bf0fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 01bf2000 -- 01bf6fff 00005000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01bfb000 -- 01bfdfff 00003000 RW     mmap
[#0 (UE) Fire(29:29)]: 01c00000 -- 01c0ffff 00010000 RW     malloc: ChunkedBlock(16)
[#0 (UE) Fire(29:29)]: 01c11000 -- 01c14fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01c1a000 -- 01c1dfff 00004000 R  S   libm.so: .text
[#0 (UE) Fire(29:29)]: 01c20000 -- 01c8ffff 00070000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 01c91000 -- 01c96fff 00006000 R  S   /bin/Fire: .text
[#0 (UE) Fire(29:29)]: 01c98000 -- 01c9dfff 00006000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01ca0000 -- 01d0ffff 00070000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 01d11000 -- 01d14fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01d16000 -- 01d18fff 00003000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01d1a000 -- 01d1bfff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 01d1d000 -- 01d1dfff 00001000 RW     mmap
[#0 (UE) Fire(29:29)]: 01d20000 -- 01d2ffff 00010000 RW     malloc: ChunkedBlock(4088)
[#0 (UE) Fire(29:29)]: 01d31000 -- 01d32fff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01d34000 -- 01d36fff 00003000 RW     mmap
[#0 (UE) Fire(29:29)]: 01d38000 -- 01d3afff 00003000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01d3c000 -- 01d3dfff 00002000 R  S   libcrypt.so: .text
[#0 (UE) Fire(29:29)]: 01d40000 -- 01d4ffff 00010000 RW     malloc: ChunkedBlock(256)
[#0 (UE) Fire(29:29)]: 01d51000 -- 01d60fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01d62000 -- 01d71fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 01d73000 -- 01d74fff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01d78000 -- 01d79fff 00002000 RW     Thread-specific
[#0 (UE) Fire(29:29)]: 01d7b000 -- 01d7cfff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 01d80000 -- 01d8ffff 00010000 RW     malloc: ChunkedBlock(128)
[#0 (UE) Fire(29:29)]: 01d91000 -- 01d92fff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01d94000 -- 01d96fff 00003000 RW     mmap
[#0 (UE) Fire(29:29)]: 01d98000 -- 01d9afff 00003000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01d9c000 -- 01d9dfff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 01da0000 -- 01daffff 00010000 RW     malloc: ChunkedBlock(256)
[#0 (UE) Fire(29:29)]: 01db1000 -- 01db5fff 00005000 R  S   libtextcodec.so: .text
[#0 (UE) Fire(29:29)]: 01db7000 -- 01dbbfff 00005000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01dbd000 -- 01dbdfff 00001000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01dc0000 -- 01dcffff 00010000 RW     malloc: ChunkedBlock(32)
[#0 (UE) Fire(29:29)]: 01dd1000 -- 01dd2fff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01dd4000 -- 01dd5fff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 01dd7000 -- 01dd8fff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 01dda000 -- 01dddfff 00004000 RW     mmap
[#0 (UE) Fire(29:29)]: 01de0000 -- 020effff 00310000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 020f1000 -- 02100fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02102000 -- 02111fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 02113000 -- 02122fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02124000 -- 02133fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 02135000 -- 02144fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0214a000 -- 02159fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 0215b000 -- 0216afff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0216c000 -- 0217bfff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 0217d000 -- 0218cfff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0218e000 -- 02192fff 00005000 R  S   /res/fonts/KaticaBold12.font
[#0 (UE) Fire(29:29)]: 02194000 -- 02198fff 00005000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0219a000 -- 0219dfff 00004000 R  S   /res/fonts/KaticaRegular10.font
[#0 (UE) Fire(29:29)]: 0219f000 -- 021a2fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 021a4000 -- 021a8fff 00005000 R  S   /res/fonts/KaticaRegular12.font
[#0 (UE) Fire(29:29)]: 021aa000 -- 021aefff 00005000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 021b0000 -- 021b3fff 00004000 R  S   /res/fonts/LizaBlack10.font
[#0 (UE) Fire(29:29)]: 021b5000 -- 021b8fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 021ba000 -- 021c3fff 0000a000 R  S   /res/fonts/LizaBlack24.font
[#0 (UE) Fire(29:29)]: 021c5000 -- 021cefff 0000a000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 021d0000 -- 021ddfff 0000e000 R  S   /res/fonts/LizaBlack36.font
[#0 (UE) Fire(29:29)]: 021df000 -- 021ecfff 0000e000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 021ee000 -- 021f1fff 00004000 R  S   /res/fonts/LizaBold10.font
[#0 (UE) Fire(29:29)]: 021f3000 -- 021f6fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 021f8000 -- 02201fff 0000a000 R  S   /res/fonts/LizaBold24.font
[#0 (UE) Fire(29:29)]: 02203000 -- 0220cfff 0000a000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0220e000 -- 0221bfff 0000e000 R  S   /res/fonts/LizaBold36.font
[#0 (UE) Fire(29:29)]: 0221d000 -- 0222afff 0000e000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0222c000 -- 0222ffff 00004000 R  S   /res/fonts/LizaRegular10.font
[#0 (UE) Fire(29:29)]: 02231000 -- 02234fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02236000 -- 0223ffff 0000a000 R  S   /res/fonts/LizaRegular24.font
[#0 (UE) Fire(29:29)]: 02241000 -- 0224afff 0000a000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0224c000 -- 02259fff 0000e000 R  S   /res/fonts/LizaRegular36.font
[#0 (UE) Fire(29:29)]: 0225b000 -- 02268fff 0000e000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0226a000 -- 0226ffff 00006000 R  S   /res/fonts/PebbletonBold14.font
[#0 (UE) Fire(29:29)]: 02271000 -- 02276fff 00006000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02278000 -- 0227dfff 00006000 R  S   /res/fonts/PebbletonRegular14.font
[#0 (UE) Fire(29:29)]: 0227f000 -- 02284fff 00006000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02288000 -- 0228bfff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02290000 -- 022affff 00020000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 022b1000 -- 022bcfff 0000c000 R  S   libttf.so: .text
[#0 (UE) Fire(29:29)]: 022c0000 -- 022dffff 00020000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 022e1000 -- 022f0fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 022f2000 -- 02301fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02312000 -- 0231dfff 0000c000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02320000 -- 0232ffff 00010000        malloc: ChunkedBlock(500)
[#0 (UE) Fire(29:29)]: 02331000 -- 02340fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 02342000 -- 02351fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02353000 -- 02362fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 02364000 -- 02373fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02376000 -- 02385fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 02387000 -- 02396fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 02398000 -- 02399fff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 0239b000 -- 0239cfff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 023a0000 -- 023affff 00010000 RW     malloc: ChunkedBlock(500)
[#0 (UE) Fire(29:29)]: 023c0000 -- 023cffff 00010000 RW     malloc: ChunkedBlock(2032)
[#0 (UE) Fire(29:29)]: 023d1000 -- 02414fff 00044000 R  S   /usr/lib/libcompress.so
[#0 (UE) Fire(29:29)]: 02420000 -- 0242ffff 00010000 RW     malloc: ChunkedBlock(64)
[#0 (UE) Fire(29:29)]: 02440000 -- 0244ffff 00010000        malloc: ChunkedBlock(32752)
[#0 (UE) Fire(29:29)]: 02460000 -- 0246ffff 00010000        malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 02480000 -- 0248ffff 00010000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 024a0000 -- 024bffff 00020000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 024c1000 -- 02688fff 001c8000 R  S   /usr/lib/libgfx.so
[#0 (UE) Fire(29:29)]: 02690000 -- 0269ffff 00010000 RW     malloc: ChunkedBlock(4088)
[#0 (UE) Fire(29:29)]: 026b0000 -- 026bffff 00010000 RW     malloc: ChunkedBlock(64)
[#0 (UE) Fire(29:29)]: 026d0000 -- 026dffff 00010000 RW     malloc: ChunkedBlock(64)
[#0 (UE) Fire(29:29)]: 026f0000 -- 026fffff 00010000 RW     malloc: ChunkedBlock(4088)
[#0 (UE) Fire(29:29)]: 02710000 -- 0271ffff 00010000 RW     malloc: ChunkedBlock(4088)
[#0 (UE) Fire(29:29)]: 02730000 -- 0273ffff 00010000 RW     malloc: ChunkedBlock(64)
[#0 (UE) Fire(29:29)]: 02750000 -- 0275ffff 00010000 RW     malloc: ChunkedBlock(2032)
[#0 (UE) Fire(29:29)]: 02770000 -- 0277ffff 00010000 RW     malloc: ChunkedBlock(4088)
[#0 (UE) Fire(29:29)]: 02790000 -- 0279ffff 00010000 RW     malloc: ChunkedBlock(64)
[#0 (UE) Fire(29:29)]: 027b0000 -- 027bffff 00010000 RW     malloc: ChunkedBlock(8184)
[#0 (UE) Fire(29:29)]: 027d0000 -- 027dffff 00010000        malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 027f0000 -- 027fffff 00010000 RW     malloc: ChunkedBlock(32)
[#0 (UE) Fire(29:29)]: 02810000 -- 0289ffff 00090000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 028a1000 -- 03278fff 009d8000 R  S   /usr/lib/libgui.so
[#0 (UE) Fire(29:29)]: 03280000 -- 0328ffff 00010000 RW     malloc: ChunkedBlock(32)
[#0 (UE) Fire(29:29)]: 03291000 -- 032a2fff 00012000 R  S   /bin/Fire
[#0 (UE) Fire(29:29)]: 032b0000 -- 032bffff 00010000 RW     malloc: ChunkedBlock(64)
[#0 (UE) Fire(29:29)]: 032d0000 -- 032dffff 00010000 RW     malloc: ChunkedBlock(16376)
[#0 (UE) Fire(29:29)]: 03330000 -- 0338ffff 00060000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 033a0000 -- 033effff 00050000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 03a60000 -- 03d0ffff 002b0000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 04513000 -- 0452afff 00018000 R  S   libgcc_s.so: .text
[#0 (UE) Fire(29:29)]: 0452c000 -- 04543fff 00018000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04545000 -- 045c9fff 00085000 R  S   libc.so: .text
[#0 (UE) Fire(29:29)]: 045cb000 -- 0464ffff 00085000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04651000 -- 04671fff 00021000 R  S   libcrypto.so: .text
[#0 (UE) Fire(29:29)]: 04673000 -- 04693fff 00021000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04695000 -- 046a9fff 00015000 R  S   libcompress.so: .text
[#0 (UE) Fire(29:29)]: 046ab000 -- 046bffff 00015000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 046c1000 -- 0470bfff 0004b000 R  S   libcore.so: .text
[#0 (UE) Fire(29:29)]: 0470d000 -- 04757fff 0004b000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04759000 -- 047c8fff 00070000 R  S   libgfx.so: .text
[#0 (UE) Fire(29:29)]: 047ca000 -- 04839fff 00070000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0483b000 -- 04841fff 00007000 R  S   libipc.so: .text
[#0 (UE) Fire(29:29)]: 04843000 -- 04849fff 00007000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0484b000 -- 0484cfff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 04850000 -- 0485ffff 00010000 RW     malloc: ChunkedBlock(1016)
[#0 (UE) Fire(29:29)]: 04861000 -- 04864fff 00004000 R  S   libpthread.so: .text
[#0 (UE) Fire(29:29)]: 04866000 -- 04869fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0486b000 -- 0486cfff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0486e000 -- 04871fff 00004000 R  S   libthread.so: .text
[#0 (UE) Fire(29:29)]: 04873000 -- 04876fff 00004000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04878000 -- 04879fff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 0487b000 -- 0487cfff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 0487e000 -- 048b5fff 00038000 R  S   libregex.so: .text
[#0 (UE) Fire(29:29)]: 048b7000 -- 048eefff 00038000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 048f0000 -- 048f1fff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 048f3000 -- 048f4fff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 048f6000 -- 048f7fff 00002000 R  S   libsyntax.so: .text
[#0 (UE) Fire(29:29)]: 048f9000 -- 048fafff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 048fc000 -- 048fdfff 00002000 RW     mmap
[#0 (UE) Fire(29:29)]: 048ff000 -- 04900fff 00002000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04902000 -- 04a9cfff 0019b000 R  S   libgui.so: .text
[#0 (UE) Fire(29:29)]: 04a9e000 -- 04c38fff 0019b000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04c3a000 -- 04c45fff 0000c000 RW     mmap
[#0 (UE) Fire(29:29)]: 04c47000 -- 04c52fff 0000c000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04c54000 -- 04c63fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 04c65000 -- 04c74fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04c80000 -- 04ccffff 00050000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 04cd1000 -- 04ce0fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 04ce2000 -- 04cf1fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04d00000 -- 04d0ffff 00010000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 04d11000 -- 04d20fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 04d22000 -- 04d31fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04d40000 -- 04d6ffff 00030000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 04d71000 -- 04d80fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 04d82000 -- 04d91fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04da0000 -- 04daffff 00010000 RW     malloc: ChunkedBlock(32752)
[#0 (UE) Fire(29:29)]: 04db1000 -- 04dc0fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 04dc2000 -- 04dd1fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04de0000 -- 04dfffff 00020000 RW     malloc: BigAllocationBlock
[#0 (UE) Fire(29:29)]: 04e01000 -- 04e10fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 04e12000 -- 04e21fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04e30000 -- 04e3ffff 00010000 RW     malloc: ChunkedBlock(16376)
[#0 (UE) Fire(29:29)]: 04e41000 -- 04e50fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 04e52000 -- 04e61fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 04e63000 -- 04e72fff 00010000 RW     mmap
[#0 (UE) Fire(29:29)]: 04e74000 -- 04e83fff 00010000 RW     MmapRegion ShadowData
[#0 (UE) Fire(29:29)]: 157bd000 -- 157f4fff 00038000 R XS   libregex.so: .text
[#0 (UE) Fire(29:29)]: 157f5000 -- 157f5fff 00001000 R      libregex.so: .relro
[#0 (UE) Fire(29:29)]: 157f6000 -- 157f6fff 00001000 RW     libregex.so: .data
[#0 (UE) Fire(29:29)]: 15a37000 -- 15a3afff 00004000 R XS   libm.so: .text
[#0 (UE) Fire(29:29)]: 15a3b000 -- 15a3bfff 00001000 R      libm.so: .relro
[#0 (UE) Fire(29:29)]: 16219000 -- 1627ffff 00067000 R XS   /usr/lib/Loader.so
[#0 (UE) Fire(29:29)]: 16280000 -- 16281fff 00002000 RW     /usr/lib/Loader.so (data-rw)
[#0 (UE) Fire(29:29)]: 17ea0000 -- 17eaefff 0000f000 R XS   libx86.so: .text
[#0 (UE) Fire(29:29)]: 17eaf000 -- 17eaffff 00001000 R      libx86.so: .relro
[#0 (UE) Fire(29:29)]: 17eb0000 -- 17ebafff 0000b000 RW     libx86.so: .data
[#0 (UE) Fire(29:29)]: 1f82a000 -- 1f874fff 0004b000 R XS   libcore.so: .text
[#0 (UE) Fire(29:29)]: 1f875000 -- 1f876fff 00002000 R      libcore.so: .relro
[#0 (UE) Fire(29:29)]: 1f877000 -- 1f877fff 00001000 RW     libcore.so: .data
[#0 (UE) Fire(29:29)]: 66fc3000 -- 67061fff 0009f000 R XS   /bin/UserspaceEmulator: .text
[#0 (UE) Fire(29:29)]: 67063000 -- 67063fff 00001000 R      /bin/UserspaceEmulator: .relro
[#0 (UE) Fire(29:29)]: 67064000 -- 67064fff 00001000 RW     /bin/UserspaceEmulator: .data
[#0 (UE) Fire(29:29)]: 69726000 -- 69727fff 00002000 R XS   libcrypt.so: .text
[#0 (UE) Fire(29:29)]: 69728000 -- 69728fff 00001000 R      libcrypt.so: .relro
[#0 (UE) Fire(29:29)]: 69729000 -- 69729fff 00001000 RW     libcrypt.so: .data
[#0 (UE) Fire(29:29)]: 6ce24000 -- 6ce27fff 00004000 R XS   libpthread.so: .text
[#0 (UE) Fire(29:29)]: 6ce28000 -- 6ce28fff 00001000 R      libpthread.so: .relro
[#0 (UE) Fire(29:29)]: 6ce29000 -- 6ce29fff 00001000 RW     libpthread.so: .data
[#0 (UE) Fire(29:29)]: 7ef89000 -- 7efa0fff 00018000 R XS   libgcc_s.so: .text
[#0 (UE) Fire(29:29)]: 7efa1000 -- 7efa1fff 00001000 RW     libgcc_s.so: .data
[#0 (UE) Fire(29:29)]: 94fd5000 -- 94fedfff 00019000 R XS   libdebug.so: .text
[#0 (UE) Fire(29:29)]: 94fee000 -- 94feefff 00001000 R      libdebug.so: .relro
[#0 (UE) Fire(29:29)]: 94fef000 -- 94feffff 00001000 RW     libdebug.so: .data
[#0 (UE) Fire(29:29)]: a1d98000 -- a1e1cfff 00085000 R XS   libc.so: .text
[#0 (UE) Fire(29:29)]: a1e1e000 -- a1e1ffff 00002000 R      libc.so: .relro
[#0 (UE) Fire(29:29)]: a1e20000 -- a1e20fff 00001000 RW     libc.so: .data
[#0 (UE) Fire(29:29)]: aee29000 -- aee49fff 00021000 R XS   libcrypto.so: .text
[#0 (UE) Fire(29:29)]: aee4a000 -- aee4afff 00001000 R      libcrypto.so: .relro
[#0 (UE) Fire(29:29)]: aee4b000 -- aee4bfff 00001000 RW     libcrypto.so: .data
[#0 (UE) Fire(29:29)]: b89ab000 -- b89abfff 00001000 R XS C libsystem.so: .text
[#0 (UE) Fire(29:29)]: b89ac000 -- b89acfff 00001000 R      libsystem.so: .relro
[#0 (UE) Fire(29:29)]: Kernel regions:
[#0 (UE) Fire(29:29)]: BEGIN       END         SIZE        ACCESS  NAME
[#0 (UE) Fire(29:29)]: c1104000 -- c1104fff 00001000 RW     HPET MMIO
[#0 (UE) Fire(29:29)]: c1106000 -- c1106fff 00001000 R    C Signal trampolines
[#0 (UE) Fire(29:29)]: c1108000 -- c1117fff 00010000 RW  T  Kernel stack (thread 0)
[#0 (UE) Fire(29:29)]: c112a000 -- c1139fff 00010000 RW  T  Kernel stack (thread 2)
[#0 (UE) Fire(29:29)]: c113b000 -- c114afff 00010000 RW  T  Kernel stack (thread 3)
[#0 (UE) Fire(29:29)]: c114c000 -- c114cfff 00001000  W     UHCI Framelist
[#0 (UE) Fire(29:29)]: c114e000 -- c114ffff 00002000  W     UHCI Queue Head Pool
[#0 (UE) Fire(29:29)]: c1151000 -- c1152fff 00002000  W     UHCI Transfer Descriptor Pool
[#0 (UE) Fire(29:29)]: c1154000 -- c1163fff 00010000 RW  T  Kernel stack (thread 4)
[#0 (UE) Fire(29:29)]: c1165000 -- c1165fff 00001000 RW     E1000 RX
[#0 (UE) Fire(29:29)]: c1167000 -- c1167fff 00001000 RW     E1000 TX
[#0 (UE) Fire(29:29)]: c1169000 -- c1188fff 00020000 RW     E1000 MMIO
[#0 (UE) Fire(29:29)]: c118a000 -- c118bfff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c118d000 -- c118efff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c1190000 -- c1191fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c1193000 -- c1194fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c1196000 -- c1197fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c1199000 -- c119afff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c119c000 -- c119dfff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c119f000 -- c11a0fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11a2000 -- c11a3fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11a5000 -- c11a6fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11a8000 -- c11a9fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11ab000 -- c11acfff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11ae000 -- c11affff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11b1000 -- c11b2fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11b4000 -- c11b5fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11b7000 -- c11b8fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11ba000 -- c11bbfff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11bd000 -- c11befff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11c0000 -- c11c1fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11c3000 -- c11c4fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11c6000 -- c11c7fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11c9000 -- c11cafff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11cc000 -- c11cdfff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11cf000 -- c11d0fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11d2000 -- c11d3fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11d5000 -- c11d6fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11d8000 -- c11d9fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11db000 -- c11dcfff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11de000 -- c11dffff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11e1000 -- c11e2fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11e4000 -- c11e5fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11e7000 -- c11e8fff 00002000 RW     E1000 RX buffer
[#0 (UE) Fire(29:29)]: c11ea000 -- c11ebfff 00002000 RW     E1000 TX buffer
[#0 (UE) Fire(29:29)]: c11ed000 -- c11eefff 00002000 RW     E1000 TX buffer
[#0 (UE) Fire(29:29)]: c11f0000 -- c11f1fff 00002000 RW     E1000 TX buffer
[#0 (UE) Fire(29:29)]: c11f3000 -- c11f4fff 00002000 RW     E1000 TX buffer
[#0 (UE) Fire(29:29)]: c11f6000 -- c11f7fff 00002000 RW     E1000 TX buffer
[#0 (UE) Fire(29:29)]: c11f9000 -- c11fafff 00002000 RW     E1000 TX buffer
[#0 (UE) Fire(29:29)]: c11fc000 -- c11fdfff 00002000 RW     E1000 TX buffer
[#0 (UE) Fire(29:29)]: c11ff000 -- c1200fff 00002000 RW     E1000 TX buffer
[#0 (UE) Fire(29:29)]: c1202000 -- c1203fff 00002000 RW     AHCI HBA
[#0 (UE) Fire(29:29)]: c1205000 -- c1205fff 00001000 RW     AHCI Port Command List
[#0 (UE) Fire(29:29)]: c1207000 -- c1207fff 00001000 RW     AHCI Port Command List
[#0 (UE) Fire(29:29)]: c1209000 -- c1209fff 00001000 RW     AHCI Port Command List
[#0 (UE) Fire(29:29)]: c120b000 -- c120bfff 00001000 RW     AHCI Port Command List
[#0 (UE) Fire(29:29)]: c120d000 -- c120dfff 00001000 RW     AHCI Port Command List
[#0 (UE) Fire(29:29)]: c120f000 -- c120ffff 00001000 RW     AHCI Port Command List
[#0 (UE) Fire(29:29)]: c1211000 -- c1211fff 00001000 RW     Ext2FS: Block group descriptors
[#0 (UE) Fire(29:29)]: c1213000 -- c3922fff 02710000 RW     KBuffer
[#0 (UE) Fire(29:29)]: c3924000 -- c395efff 0003b000 RW     KBuffer
[#0 (UE) Fire(29:29)]: c3960000 -- c3960fff 00001000 RW     Ext2FS: Cached bitmap block
[#0 (UE) Fire(29:29)]: c3962000 -- c3962fff 00001000 RW     Ext2FS: Cached bitmap block
[#0 (UE) Fire(29:29)]: c3964000 -- c3973fff 00010000 RW  T  Kernel stack (thread 5)
[#0 (UE) Fire(29:29)]: c3975000 -- c3984fff 00010000 RW  T  Kernel stack (thread 6)
[#0 (UE) Fire(29:29)]: c3986000 -- c3995fff 00010000 RW     Kernel Packet Buffer
[#0 (UE) Fire(29:29)]: c1119000 -- c1119fff 00001000 RW     Ext2FS: Cached bitmap block
[#0 (UE) Fire(29:29)]: c3997000 -- c39b6fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c39b8000 -- c39d7fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c39d9000 -- c39f8fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c39fa000 -- c3a19fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3a1b000 -- c3a3afff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3a3c000 -- c3a5bfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3a5d000 -- c3a7cfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3a7e000 -- c3a9dfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3a9f000 -- c3abefff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3ac0000 -- c3adffff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3ae1000 -- c3b00fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3b02000 -- c3b21fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3b23000 -- c3b42fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3b44000 -- c3b63fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3b65000 -- c3b84fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3b86000 -- c3ba5fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3ba7000 -- c3bc6fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3bc8000 -- c3be7fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3be9000 -- c3c08fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3c0a000 -- c3c29fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3c2b000 -- c3c4afff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3c4c000 -- c3c6bfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3c6d000 -- c3c8cfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3c8e000 -- c3cadfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3caf000 -- c3ccefff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3cd0000 -- c3ceffff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3cf1000 -- c3d10fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3d12000 -- c3d21fff 00010000 RW  T  Kernel stack (thread 8)
[#0 (UE) Fire(29:29)]: c3d23000 -- c3d32fff 00010000 RW  T  Kernel stack (thread 9)
[#0 (UE) Fire(29:29)]: c3d34000 -- c3d43fff 00010000 RW  T  Kernel stack (thread 10)
[#0 (UE) Fire(29:29)]: c3d45000 -- c3d54fff 00010000 RW  T  Kernel stack (thread 11)
[#0 (UE) Fire(29:29)]: c3d56000 -- c3d65fff 00010000 RW  T  Kernel stack (thread 12)
[#0 (UE) Fire(29:29)]: c3d67000 -- c3d76fff 00010000 RW  T  Kernel stack (thread 13)
[#0 (UE) Fire(29:29)]: c3d78000 -- c3d87fff 00010000 RW  T  Kernel stack (thread 14)
[#0 (UE) Fire(29:29)]: c3d89000 -- c3d98fff 00010000 RW  T  Kernel stack (thread 15)
[#0 (UE) Fire(29:29)]: c3d9a000 -- c3da9fff 00010000 RW  T  Kernel stack (thread 16)
[#0 (UE) Fire(29:29)]: c3dab000 -- c3dbafff 00010000 RW  T  Kernel stack (thread 17)
[#0 (UE) Fire(29:29)]: c3dbc000 -- c3dcbfff 00010000 RW  T  Kernel stack (thread 18)
[#0 (UE) Fire(29:29)]: c3dcd000 -- c3ddcfff 00010000 RW  T  Kernel stack (thread 19)
[#0 (UE) Fire(29:29)]: c3dde000 -- c3dedfff 00010000 RW  T  Kernel stack (thread 20)
[#0 (UE) Fire(29:29)]: c3def000 -- c3dfefff 00010000 RW  T  Kernel stack (thread 21)
[#0 (UE) Fire(29:29)]: c3e00000 -- c3e0ffff 00010000 RW  T  Kernel stack (thread 22)
[#0 (UE) Fire(29:29)]: c3e11000 -- c3e20fff 00010000 RW  T  Kernel stack (thread 23)
[#0 (UE) Fire(29:29)]: c3e22000 -- c3e41fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3e43000 -- c3e62fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3e64000 -- c3e83fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c3e85000 -- c4284fff 00400000 RW     KBuffer
[#0 (UE) Fire(29:29)]: c4286000 -- c42a5fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c42a7000 -- c42c6fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c111b000 -- c111bfff 00001000 RW     KBuffer
[#0 (UE) Fire(29:29)]: c42c8000 -- c42e7fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c42e9000 -- c4308fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c430a000 -- c4329fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c432b000 -- c434afff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c434c000 -- c436bfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c436d000 -- c438cfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c438e000 -- c43adfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c43af000 -- c43cefff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c43d0000 -- c43effff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c43f1000 -- c4410fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4412000 -- c4431fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4433000 -- c4452fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4454000 -- c4473fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4475000 -- c4494fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4496000 -- c44b5fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c44b7000 -- c44d6fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c44d8000 -- c44f7fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c44f9000 -- c4518fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c451a000 -- c4539fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c453b000 -- c455afff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c455c000 -- c457bfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c457d000 -- c459cfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c459e000 -- c45bdfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c45bf000 -- c45defff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c45e0000 -- c45fffff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4601000 -- c4620fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4622000 -- c4641fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4643000 -- c4662fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4664000 -- c4683fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4685000 -- c46a4fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c46a6000 -- c46c5fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c46c7000 -- c46e6fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c46e8000 -- c4707fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4709000 -- c4728fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c472a000 -- c4749fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c474b000 -- c476afff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c476c000 -- c478bfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c478d000 -- c47acfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c47ae000 -- c47cdfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c47cf000 -- c47eefff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c47f0000 -- c480ffff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4811000 -- c4830fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4832000 -- c4851fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4853000 -- c4872fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4874000 -- c4893fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4895000 -- c48b4fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c48b6000 -- c48d5fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c48d7000 -- c48f6fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c48f8000 -- c4917fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4919000 -- c4938fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c493a000 -- c4959fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c495b000 -- c497afff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c497c000 -- c499bfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c499d000 -- c49bcfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c49be000 -- c49ddfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c49df000 -- c49fefff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4a00000 -- c4a1ffff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4a21000 -- c4a40fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4a42000 -- c4a51fff 00010000 RW  T  Kernel stack (thread 24)
[#0 (UE) Fire(29:29)]: c4a53000 -- c4a72fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4a74000 -- c4a93fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4a95000 -- c4ab4fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4ab6000 -- c4ad5fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4ad7000 -- c4af6fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4af8000 -- c4b17fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4b19000 -- c4b38fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4b3a000 -- c4b59fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4b5b000 -- c4b7afff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4b7c000 -- c4b9bfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4b9d000 -- c4bbcfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4bbe000 -- c4bddfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4bdf000 -- c4bfefff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4c00000 -- c4c1ffff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4c21000 -- c4c40fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4c42000 -- c4c61fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4c63000 -- c4c82fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4c84000 -- c4ca3fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4ca5000 -- c4cc4fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4cc6000 -- c4ce5fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4ce7000 -- c4d06fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4d08000 -- c4d27fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4d29000 -- c4d48fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4d4a000 -- c4d69fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4d6b000 -- c4d8afff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4d8c000 -- c4dabfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4dad000 -- c4dccfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4dce000 -- c4dedfff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4def000 -- c4e0efff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4e10000 -- c4e2ffff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4e31000 -- c4e50fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4e52000 -- c4e61fff 00010000 RW  T  Kernel stack (thread 25)
[#0 (UE) Fire(29:29)]: c4e74000 -- c4e93fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4e95000 -- c4eb4fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4eb6000 -- c4ed5fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4ed7000 -- c4ef6fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c4ef8000 -- c4f17fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c111d000 -- c111dfff 00001000 RW     Ext2FS: Cached bitmap block
[#0 (UE) Fire(29:29)]: c534c000 -- c544bfff 00100000 RW     kmalloc subheap
[#0 (UE) Fire(29:29)]: c1001000 -- c1100fff 00100000 RW     kmalloc subheap
[#0 (UE) Fire(29:29)]: c4f19000 -- c5318fff 00400000 RW     KBuffer
[#0 (UE) Fire(29:29)]: c544d000 -- c584cfff 00400000 RW     KBuffer
[#0 (UE) Fire(29:29)]: c584e000 -- c585dfff 00010000 RW  T  Kernel stack (thread 29)
[#0 (UE) Fire(29:29)]: c531a000 -- c5339fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c585f000 -- c587efff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c5880000 -- c589ffff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c58a1000 -- c58c0fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c58c2000 -- c58e1fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c58e3000 -- c5902fff 00020000 RW     DoubleBuffer
[#0 (UE) Fire(29:29)]: c5904000 -- c5923fff 00020000 RW     DoubleBuffer
[#0 FinalizerTask(3:3)]: Generating coredump for pid: 29
[FinalizerTask(3:3)]: RTC: Year: 2021, month: 3, day: 14, hour: 7, minute: 24, second: 47
CrashDaemon(20): New coredump file: /tmp/coredump/(UE) Fire_29_1615706687
Shell(25): Job entry 'UserspaceEmulator Fire' deleted in 7135 ms
CrashDaemon(20): --- Backtrace for thread #0 (TID 29) ---
CrashDaemon(20): 0xb89ab188: [libsystem.so]  +0x8 (syscall.cpp:35)
CrashDaemon(20): 0xa1dacc96: [libc.so] __assertion_failed +0x106 (syscall.h:44)
CrashDaemon(20): 0x67057698: [/bin/UserspaceEmulator] UserspaceEmulator::SoftMMU::write32(X86::LogicalAddress, UserspaceEmulator::ValueWithShadow<unsigned int>) +0x108 (SoftMMU.cpp:227)
CrashDaemon(20): 0x6702930d: [/bin/UserspaceEmulator] UserspaceEmulator::SoftCPU::write_memory32(X86::LogicalAddress, UserspaceEmulator::ValueWithShadow<unsigned int>) +0x3d (SoftCPU.cpp:197)
CrashDaemon(20): 0x6702c674: [/bin/UserspaceEmulator] UserspaceEmulator::SoftCPU::MOV_RM32_reg32(X86::Instruction const&) +0xb4 (Instruction.h:686)
CrashDaemon(20): 0x66fc9ecb: [/bin/UserspaceEmulator] UserspaceEmulator::Emulator::exec() +0x48b (Emulator.cpp:251)
CrashDaemon(20): 0x66fc7253: [/bin/UserspaceEmulator] main +0x723 (main.cpp:81)
CrashDaemon(20): 0x66fc7445: [/bin/UserspaceEmulator] _start +0x45 (crt0_shared.cpp:60)
```
</p>
</details>

For GUI applications, `ue` is crashing in parsing compressed PNG files.

```
UserspaceEmulator(47): interpreter: /usr/lib/Loader.so
SoftMMU::write32: No region for @ 0x0ffffa2c
=={47}==    0xb3c40fe2  [libcompress.so]: Compress::CanonicalCode::from_bytes(AK::Span<unsigned char const>) +0x22 (Deflate.cpp:68)
=={47}==    0xb3c4327e  [libcompress.so]: Compress::DeflateDecompressor::decode_codes(Compress::CanonicalCode&, AK::Optional<Compress::CanonicalCode>&) +0x7be (Deflate.cpp:393)
=={47}==    0xb3c44f41  [libcompress.so]: Compress::DeflateDecompressor::read(AK::Span<unsigned char>) [clone .part.0] +0xc61 (Deflate.cpp:250)
=={47}==    0xb3c45d28  [libcompress.so]: Compress::DeflateDecompressor::decompress_all(AK::Span<unsigned char const>) +0x248 (Span.h:89)
=={47}==    0x8a574e0d  [libgfx.so]: Gfx::decode_png_bitmap(Gfx::PNGLoadingContext&) +0x9d (Optional.h:127)
=={47}==    0x8a5763b9  [libgfx.so]: Gfx::load_png_impl(unsigned char const*, unsigned long) +0xc9 (PNGLoader.cpp:817)
=={47}==    0x8a576766  [libgfx.so]: Gfx::load_png(AK::StringView const&) +0xd6 (Atomic.h:266)
=={47}==    0x536e4189  [/usr/Tests/LibGfx/image-decoder]: main +0x5e9 (image-decoder.cpp:131)
=={47}==    0x536e45f5  [/usr/Tests/LibGfx/image-decoder]: _start +0x45 (crt0_shared.cpp:60)
=={47}==    0x08028aa2
=={47}==    0x0800155b
UserspaceEmulator(47): USERSPACE(47) ASSERTION FAILED: false
../Userland/DevTools/UserspaceEmulator/SoftMMU.cpp:227
ASSERTION FAILED: false
```

**Comment by IdanHo** [comment]
As far as i can tell this is just the UE emulated program running out of stack space. (my changes CanonicalCode made it 512 bytes larger on the stack) ill look into optimizing that in CanonicalCode but just increasing the stack space for UE programs by a couple KiBs might be beneficial (https://github.com/SerenityOS/serenity/blob/master/Userland/DevTools/UserspaceEmulator/Emulator.cpp#L50)

**Comment by bcoles** [comment]
> increasing the stack space for UE programs by a couple KiBs might be beneficial

640K ought to be enough for anyone, although perhaps this should be bumped up to match the userspace thread stack size:

https://github.com/SerenityOS/serenity/blob/304e1938362636eeebc5f94d424ecbc4e9cd7072/Kernel/Thread.h#L1066-L1067

**Comment by IdanHo** [comment]
> 640K ought to be enough for anyone
640K would be more than enough, but its currently 64K

**Comment by bcoles** [comment]
https://strawpoll.com/gyzhjxq19

**Comment by IdanHo** [comment]
You should post that poll in IRC/a new issue since this one will be closed once the PR is merged

**Comment by bcoles** [comment]
Closed in #5785.
